### ISSUE TITLE
Use the hashed v2 lookup API for 3PIDs

### ISSUE THREAD

**Issue by lampholder** [vector-im/riot-web#10556]

**Comment by lampholder** [comment]
There's a `hash_details` endpoint in synapse that riot can call to see which hashing algorithms the server supports - Riot should use `sha256` if the server supports it, otherwise it should use `none` if that's all the server supports (we should think about whether `none` requires any sort of special warning, though we only expect it to be used in niche cases).

**Comment by jryans** [comment]
Riot Web recently added a usage of `v1/bulk_lookup` to support many lookups in Settings, but this can be replaced by `v2/lookup` (defined by this hashing MSC) which supports many lookups in one request.

**Comment by turt2live** [comment]
Conclusions from the syncs today:
* Theoretically we can drop lookup support in Riot, however we won't do that yet.
* In future we can consider a 'bind status' API to help us in the areas where we currently abuse lookups (https://github.com/matrix-org/matrix-doc/issues/2239 | disconnecting IS, checking bindings, verifying a third party invite belonging to us)
* We will keep using a lookup in AddressPicker for UX. We want display names and avatars if possible.

**Comment by turt2live** [comment]
Implementation is done solely in the js-sdk: https://github.com/matrix-org/matrix-js-sdk/pull/1021

Currently blocked on having an identity server to test against that doesn't 500 the request.

**Comment by turt2live** [comment]
unblocked: we fixed it
